modules/nextflow/src/main/groovy/nextflow/processor/LocalPollingMonitor.groovy

```diff
@@ -128,6 +143,15 @@ class LocalPollingMonitor extends TaskPollingMonitor {
         (session.getExecConfigProp(name, 'memory', OS.getTotalPhysicalMemorySize()) as MemoryUnit).toBytes()
     }
 
+    @PackageScope
+    static long configDisk(Session session, String name) {
+        if( session.workDir.fileSystem != FileSystems.default ) {
+            log.debug "Local executor is using a remote work directory -- task disk requirements will be ignored"
+            return 0
+        }
+        (session.getExecConfigProp(name, 'disk', session.workDir.toFile().getUsableSpace()) as MemoryUnit).toBytes()
```

Comment:
But the avail disk changes over time, how to take into account free spaces can increase or decrease while the workflow is running?